main.py

```diff
@@ -0,0 +1,420 @@
+#!/usr/bin/env python3
+import argparse
+import http.server
+import io
+import json
+import logging
+import re
+import shutil
+import subprocess
+import sys
+import xml.etree.ElementTree
+
+from http import HTTPStatus
+from pathlib import Path
+from urllib.parse import parse_qs, urlparse
+
+
+LOG = logging.getLogger('')
+
+
+def parse_args():
+    p = argparse.ArgumentParser(description=__doc__)
+    p.add_argument('--listen-address', default='0.0.0.0')
+    p.add_argument('--listen-port', default=8080)
+    p.add_argument('--allow-ip', default=[], action='append')
+    p.add_argument('base_dir')
+    return p.parse_args()
+
+
+class MKVTags(object):
+    @classmethod
+    def fromstring(cls, s):
+        root = xml.etree.ElementTree.fromstring(s)
+        return cls(root)
+
+    def __init__(self, root):
+        self.root = root
+
+    def tmdb_id(self):
+        for tmdb_tag in self.root.findall(".//Tag/Targets/TargetTypeValue[.='50']/../../Simple/Name[.='TMDB']/../String"):
+            return tmdb_tag.text
+
+
+class Library(object):
+    def __init__(self, base_dir):
+        self.base_dir = base_dir
+        self.movies = {}
+
+    def scan(self):
+        movies = {}
+
+        for mkv_path in Path(self.base_dir).rglob('*.mkv'):
+            # Extract the MKV tags xml
+            extract_result = subprocess.run(['mkvextract', mkv_path.as_posix(), 'tags', '/dev/stdout'], capture_output=True)
```

Comment:
I guess this there should be some reference to how to install this on your computer in the README.

https://mkvtoolnix.download/downloads.html

Reply:
Indeed the docker container is the current way. Once we get this merged in I can get the container building and on github images and then make a README mentioning the container image.

Reply:
https://github.com/duckflix/media-server/issues/2